test/UnitSession.cpp

```diff
@@ -232,6 +233,122 @@ UnitBase::TestResult UnitSession::testSlideShow()
     return TestResult::Ok;
 }
 
+/// Load a document on server and download it multiple to this client using 2 reused channels:
+/// - wsd-session for commands (websocket)
+/// - http-session to download it, as well as to load the favicon
+UnitBase::TestResult UnitSession::testSlideShowMultiDL()
+{
+    setTestname(__func__);
+    TST_LOG("Starting Test: " << testname);
+    const int dlCount = 10;
+    int dlIter = -1;
+    try
+    {
+        std::string documentPath, documentURL;
+        std::string response;
+        helpers::getDocumentPathAndURL("setclientpart.odp", documentPath, documentURL, testname);
+
+        // Reused http wsd session + socket-poll to send commands to server
+        std::shared_ptr<SocketPoll> wsdSocketPoll = std::make_shared<SocketPoll>(testname);
+        wsdSocketPoll->startThread();
+        std::shared_ptr<http::WebSocketSession> wsdSession = helpers::loadDocAndGetSession(
+            wsdSocketPoll, Poco::URI(helpers::getTestServerURI()), documentURL, testname);
+
+        // Reused http download-session for document and favicon download from server
+        std::unique_ptr<Poco::Net::HTTPClientSession> dlSession(
+            helpers::createSession(Poco::URI(helpers::getTestServerURI())));
+        dlSession->setKeepAlive(true);
+
+        for( dlIter=0; dlIter < dlCount; ++dlIter ) {
+            // Still connected on sub-sequent commands and downloads?
+            if( dlIter > 0 ) {
+                LOK_ASSERT_EQUAL(true, wsdSession->isConnected());
+                LOK_ASSERT_EQUAL(true, dlSession->connected());
+            }
+
+            // download favicon
+            {
+                const std::string path = "/favicon.ico";
+                Poco::Net::HTTPRequest requestICO(Poco::Net::HTTPRequest::HTTP_GET, path);
+                TST_LOG("Favicon requesting from " << path);
+                dlSession->sendRequest(requestICO);
+
+                Poco::Net::HTTPResponse responseICO;
+                std::istream& rs = dlSession->receiveResponse(responseICO);
+                LOK_ASSERT_EQUAL(Poco::Net::HTTPResponse::HTTP_OK /* 200 */, responseICO.getStatus());
+                LOK_ASSERT_EQUAL(std::string("image/vnd.microsoft.icon"), responseICO.getContentType());
```

Comment:
We should assert the non-presence of the "Connection: close" header I think.

Reply:
`Connection: close` in the http header seems to be for client request only, signaling the server that no 'keep alive' is desired. 
While such a server response doesn't seem to be viable or possible here, we might want to use it in the WSD request handling to keep the connection open (as per this CR/PR) IFF not set. 
Hence we would still close the socket in the WSD if client optionally requests `Connection: close`, I earmark this and check with the WSD ClientRequestDispatcher code.

Reply:
note: this unit tests checks whether the 2 sockets are still open in subsequent requests.